### ISSUE TITLE
[🐛 Bug]: Getting connection timed out while connecting to remote webdriver through proxy configuration

### ISSUE THREAD

**Issue by satya081** [SeleniumHQ/selenium#10542]
### What happened?

I had set up a selenium grid using docker container in ec2 linux instance. please find my docker-compose.yml file as below.
###############################################################################
services:
  chrome:
    image: selenium/node-chrome:4.1.3-20220405
    ports:
      - "5900"
      - "7900"
    shm_size: 2gb
    depends_on:
      - selenium-hub
    environment:
      - SE_EVENT_BUS_HOST=selenium-hub
      - SE_EVENT_BUS_PUBLISH_PORT=4442
      - SE_EVENT_BUS_SUBSCRIBE_PORT=4443
      - SE_NODE_MAX_SESSIONS=6
      - VNC_NO_PASSWORD=1
      - SE_SESSION_REQUEST_TIMEOUT=500
      - JAVA_OPTS=-Dwebdriver.chrome.whitelistedIps=
      - SE_NODE_OVERRIDE_MAX_SESSIONS=true
  selenium-hub:
    image: selenium/hub:4.1.3-20220405
    container_name: selenium-hub
    ports:
      - "2000:4442"
      - "2001:4443"
      - "2002:4444"
    environment:
      - GRID_MAX_SESSION=6
      - GRID_BROWSER_TIMEOUT=6000
      - GRID_TIMEOUT=6000
      - JAVA_OPTS=-Dwebdriver.chrome.whitelistedIps=
###############################################################################
once selenium grid is ready, I am trying to run my automation suite from another machine(windows 10 from intellij) and connecting to this selenium grid by proxy configuration(the solution mentioned in https://github.com/SeleniumHQ/selenium/issues/10231)

                   ChromeOptions options = new ChromeOptions();
                    options.addArguments("--incognito");
                    options.addArguments("--disable-gpu");
                    options.addArguments("--no-sandbox");
                    options.addArguments("--disable-browser-side-navigation");
                    options.setAcceptInsecureCerts(true);
                    options.setPageLoadStrategy(PageLoadStrategy.EAGER);
                    Map<String, Object> prefs = new HashMap<>();
                    prefs.put("download.default_directory", config.getProperty("path.remote.download"));
                    options.setExperimentalOption("prefs", prefs);
                    options.setExperimentalOption("excludeSwitches", new String[]{"enable-automation"});
                    capability = new DesiredCapabilities();
                    capability.setCapability(ChromeOptions.CAPABILITY, options);
                    capability.setBrowserName(CHROME);
                    capability.setPlatform(Platform.ANY);
                    capability.setCapability("ignoreZoomSetting", true);
                    options.merge(capability);
                    Proxy proxy = new Proxy(Proxy.Type.HTTP, new InetSocketAddress("XXXX", 8012));
                    ClientConfig config = ClientConfig.defaultConfig()
                            .baseUrl(new URL("YYY"))
                            .authenticateAs(new UsernameAndPassword("username", "password"))
                            .proxy(proxy);
                    driver = RemoteWebDriver.builder()
                            .oneOf(options)
                            .config(config)
                            .build();
But i am getting error as below

2022-04-13 15:07:14 DEBUG NetUtilInitializations:129 - Loopback interface: lo (Software Loopback Interface 1, 127.0.0.1)
2022-04-13 15:07:14 DEBUG NetUtil:191 - Failed to get SOMAXCONN from sysctl and file \proc\sys\net\core\somaxconn. Default: 200
2022-04-13 15:07:15 DEBUG DefaultChannelId:101 - -Dio.netty.machineId: 38:68:93:ff:fe:bb:04:31 (auto-detected)
2022-04-13 15:07:15 DEBUG AbstractByteBuf:63 - -Dio.netty.buffer.checkAccessible: true
2022-04-13 15:07:15 DEBUG AbstractByteBuf:64 - -Dio.netty.buffer.checkBounds: true
2022-04-13 15:07:15 DEBUG ResourceLeakDetectorFactory:196 - Loaded default ResourceLeakDetector: io.netty.util.ResourceLeakDetector@372d56fc
2022-04-13 15:07:15 TRACE LoggingHandler:142 - [id: 0x934f42f4] REGISTERED
2022-04-13 15:07:15 TRACE LoggingHandler:142 - [id: 0x934f42f4] CONNECT: /10.178.50.179:2002
2022-04-13 15:07:25 DEBUG NettyConnectListener:166 - Trying to recover from failing to connect channel [id: 0x934f42f4] with a retry value of true 
2022-04-13 15:07:25 DEBUG NettyConnectListener:176 - Failed to recover from connect exception: io.netty.channel.ConnectTimeoutException: connection timed out: /23.178.67.179:2002 with channel [id: 0x934f42f4]
2022-04-13 15:07:25 DEBUG AsyncCompletionHandler:71 - connection timed out: /23.178.67.179:2002
java.net.ConnectException: connection timed out: /23.178.67.179:2002
	at org.asynchttpclient.netty.channel.NettyConnectListener.onFailure(NettyConnectListener.java:179)
	at org.asynchttpclient.netty.channel.NettyChannelConnector$1.onFailure(NettyChannelConnector.java:108)
	at org.asynchttpclient.netty.SimpleChannelFutureListener.operationComplete(SimpleChannelFutureListener.java:28)
	at org.asynchttpclient.netty.SimpleChannelFutureListener.operationComplete(SimpleChannelFutureListener.java:20)
	at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:578)
	at io.netty.util.concurrent.DefaultPromise.notifyListeners0(DefaultPromise.java:571)
	at io.netty.util.concurrent.DefaultPromise.notifyListenersNow(DefaultPromise.java:550)
	at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:491)
	at io.netty.util.concurrent.DefaultPromise.setValue0(DefaultPromise.java:616)
	at io.netty.util.concurrent.DefaultPromise.setFailure0(DefaultPromise.java:609)
	at io.netty.util.concurrent.DefaultPromise.tryFailure(DefaultPromise.java:117)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe$1.run(AbstractNioChannel.java:262)
	at io.netty.util.concurrent.PromiseTask.runTask(PromiseTask.java:98)
	at io.netty.util.concurrent.ScheduledFutureTask.run(ScheduledFutureTask.java:170)
	at io.netty.util.concurrent.AbstractEventExecutor.safeExecute(AbstractEventExecutor.java:164)
	at io.netty.util.concurrent.SingleThreadEventExecutor.runAllTasks(SingleThreadEventExecutor.java:469)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:503)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:986)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.lang.Thread.run(Thread.java:748)
Caused by: io.netty.channel.ConnectTimeoutException: connection timed out: /23.178.67.179:2002
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe$1.run(AbstractNioChannel.java:261)
	... 9 more
2022-04-13 15:07:25 TRACE LoggingHandler:142 - [id: 0x934f42f4] CLOSE
2022-04-13 15:07:26 TRACE LoggingHandler:142 - [id: 0x934f42f4] UNREGISTERED
      org.openqa.selenium.SessionNotCreatedException: Could not start a new session. Response code 500. Message: java.net.ConnectException: connection timed out: /10.178.50.179:2002
Build info: version: '4.1.3', revision: '7b1ebf28ef'
System info: host: '02DI20212070173', ip: '10.153.202.119', os.name: 'Windows 10', os.arch: 'amd64', os.version: '10.0', java.version: '1.8.0_181'
Driver info: driver.version: unknown
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:126)
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:84)
	at org.openqa.selenium.remote.RemoteWebDriverBuilder.getRemoteDriver(RemoteWebDriverBuilder.java:395)
	at org.openqa.selenium.remote.RemoteWebDriverBuilder.build(RemoteWebDriverBuilder.java:368)
	at europa.edit.util.WebDriverFactory.remoteDriver(WebDriverFactory.java:171)
	at europa.edit.util.WebDriverFactory.setWebDriver(WebDriverFactory.java:73)

### How can we reproduce the issue?
by using my code mentioned above

### Operating System

Windows 10, Linux

### Selenium version

4.1.3

### What are the browser(s) and version(s) where you see this issue?

v.100.0

### What are the browser driver(s) and version(s) where you see this issue?

selenium grid using docker service

### Are you using Selenium Grid?

4.1.3

**Comment by diemol** [comment]
Duplicate of #10531

Fix will be released in the next 2 weeks.
